### DISCUSSION TITLE
changing cursor column or using \r

### DISCUSSION THREAD

**Discussion by ordo-n** [jcubic/jquery.terminal#953] [General]
Hi is there a way to use \r to go back to the start of the line, or a change cursor feature that i can use to go back to the start of the line?

I would like this to be able to draw an updating progress bar in the terminal

**Comment by jcubic** [comment]
All `\r` are removed. Do you want to go into the beginning of the line if you echo with `newline: false` or when you type something and didn't press enter?

**Comment by ordo-n** [comment]
in my case the terminal is paused so no input can occur.

i would like to echo with newline:false (to print the start of a whole progress bar) then go back to the start of the line and do it again (in order to print the next stage of the moving progress bar) until the progress bar reaches 100%

and then i can newline and resume input

the only way i can think to do it at the moment is possibly echo a span raw and change it with innerHTML until the effect is complete

perhaps that is the solution (i will give it a try when i get round to it) but it would be nice if there was a configuration to parse the return character else a function that returns to the start of the line (and like windows or linux terminals subsequent echo overwrites the line from the start)

**Reply by jcubic** [reply]
So you don't need to move the cursor at all. You need to use `term.update(-1, 'new string');` to change the last line.

**Reply by ordo-n** [reply]
perfect! thankyou!
